pkg/scaling/cache/scalers_cache.go

```diff
@@ -73,6 +89,8 @@ func (c *ScalersCache) GetPushScalers() []scalers.PushScaler {
 
 // Close closes all scalers in the cache
 func (c *ScalersCache) Close(ctx context.Context) {
+	c.mutex.RLock()
+	defer c.mutex.RUnlock()
```

Comment:
I think this needs the lock for write because `s.Scalers` is written into a few lines below
https://github.com/kedacore/keda/blob/a072763f73b6335217938b70b59c4d36e9dcf20b/pkg/scaling/cache/scalers_cache.go#L95

Reply:
and technically, right after the `c.Scalers = nil` assignment, it could `c.mutex.Unlock()` because the closing happens outside of the shared memory (which is now nil)

Reply:
****. nice catch!